internal/parse/parser.go

```diff
@@ -187,57 +209,143 @@ func isNameByte(c byte) bool {
 		'0' <= c && c <= '9' || c == '_'
 }
 
-type parseHelperTest struct {
-	bytef    func(byte) bool
-	stringf  func(string) bool
-	stringf0 func() bool
-	result   []bool
-	input    string
-	data     []string
+// These functions attempt to parse some construct, they return a bool and that
+// construct, if they n't parse they return false, restore the parser and leave
+// the default value in  other return type
+
+func (p *Parser) parseIdentifier() (string, bool) {
+	p.skipSpaces()
+	if p.pos >= len(p.input) {
+		return "", false
+	}
+	if p.peekByte('*') {
+		p.pos++
+		return "*", true
+	}
+
+	idStart := p.pos
+	if !isNameByte(p.input[p.pos]) {
+		return "", false
+	}
+	var i int
+	for i = p.pos; i < len(p.input); i++ {
+		if !isNameByte(p.input[i]) {
+			break
+		}
+	}
+	p.pos = i
+	return p.input[idStart:i], true
+}
+
+// Parses a column name or a Go type name. If parsing a Go type name then
+// struct name is in FullName.Prefix and the field name (if extant) is in
+// FullName.Name.
+// When parsing a column the table name (if extant) is in FullName.Prefix and
+// the column name is in FullName.Name func (p *Parser)
+func (p *Parser) parseFullName(idc idClass) (FullName, bool) {
+	cp := p.save()
+	var fn FullName
+	p.skipSpaces()
+	if id, ok := p.parseIdentifier(); ok {
+		fn.Prefix = id
+		if p.skipByte('.') {
+			if id, ok := p.parseIdentifier(); ok {
```

Comment:
This makes it look like the skipSpaces at the beginning of parseIdentifier is wrong. We probably don't want spaces here?

Reply:
True